### ISSUE TITLE
Unreliable CONTEXT/PASSWORD encryption in VM/template

### ISSUE THREAD

**Issue by vholer** [OpenNebula/one#5559]
**Description**

It can be observed broken or unreliable encryption of virtual machine (template) part of `CONTEXT/PASSWORD` (resp. handling of `VM_ENCRYPTED_ATTR`), during which password is not encrypted on VM/template creation, or is spontaneously decrypted later.

**To Reproduce**

#### Case 1. VM template `CONTEXT/PASSWORD` not encrypted on template creation

Example:

- Having a simple template

```
# cat template.txt 
NAME = "test"
MEMORY = "1"
VCPU = "1"
CPU = "1"
CONTEXT = [
  PASSWORD = "test"
]
```

- Password is not encrypted on template creationg

```
# onetemplate create template.txt 
ID: 2828
# onetemplate show 2828
...
TEMPLATE CONTENTS                                                               
CONTEXT=[
  PASSWORD="test" ]
CPU="1"
MEMORY="1"
VCPU="1"
```

- But, password is encrypted on template update

```
# onetemplate update 2828 template.txt
# onetemplate show 2828
...
TEMPLATE CONTENTS                                                               
CONTEXT=[
  PASSWORD="mU18gesj9B80f/iOrZrg8A==" ]
CPU="1"
MEMORY="1"
NAME="test"
VCPU="1"
```

#### Case 2. VM metadata with unencrypt CONTEXT/PASSWORD on instantiation

Example:

- Instantiate VM from same template, see PASSWORD gets unencrypted !!!**after a while**!!!

```
# cat template.txt 
NAME = "test"
MEMORY = "1"
VCPU = "1"
CPU = "1"
CONTEXT = [
  PASSWORD = "test"
]
# onetemplate create template.txt 
ID: 2831
# onetemplate show 2831 | grep PASSWORD
  PASSWORD="test" ]
# onetemplate instantiate 2831
VM ID: 2830

# while true; do onevm show 2830 | grep PASSWORD; sleep 1; done
  PASSWORD="vMku31tvO4fhVej9Az3qHg==",
  PASSWORD="vMku31tvO4fhVej9Az3qHg==",
  PASSWORD="vMku31tvO4fhVej9Az3qHg==",
  PASSWORD="vMku31tvO4fhVej9Az3qHg==",
  PASSWORD="test",
  PASSWORD="test",
  PASSWORD="test",
  PASSWORD="test",
  PASSWORD="test",
^C
```

- On any template update (e.g., due to REPORT_READY), the password gets encrypted again

```
# touch nothing
# onevm show 2830 | grep PASSWORD
  PASSWORD="test",
# touch nothing
# onevm update 2830 --append nothing
# onevm show 2830 | grep PASSWORD
  PASSWORD="vMku31tvO4fhVej9Az3qHg==",
```

**Expected behavior**
1. OpenNebula reliably encrypts on creation what's in `VM_ENCRYPTED_ATTR`
2. Once encrypted, forever encrypted.

**Details**
 - Affected Component: oned
 - Version: 5.12.10+

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by redundancydisorder** [comment]
Also seeing this on 6.0.0.2 when editing a template in sunstone

**Comment by paczerny** [comment]
[PR src](https://github.com/OpenNebula/one-ee/pull/1543)
[PR docs](https://github.com/OpenNebula/docs/pull/2041)

**Comment by paczerny** [comment]
Fixed also VN encrypted attributes on creating a VN template
[PR for tests](https://github.com/OpenNebula/development/pull/1675)

**Comment by paczerny** [comment]
Fixed also Image encryption on create
[PR src](https://github.com/OpenNebula/one-ee/pull/1614)
[PR tests](https://github.com/OpenNebula/development/pull/1678)

**Comment by dont32** [comment]
@feldsam i also have the problem with opennebula 6.4. Have you fixed?

**Comment by rsmontero** [comment]
Are you having the issue when setting password from user_inputs?

**Comment by feldsam** [comment]
> Are you having the issue when setting password from user_inputs?

Yes, exactly

**Comment by rsmontero** [comment]
Fix as per #1b70b07c2b7be4c9e58e0e661f65d2977d6e50ae

THANKS!

**Comment by feldsam** [comment]
Hello @rsmontero , thank you for fix. I tested it and password is decrypted in context.sh file.

There is still problem with guacamole, RDP connection can not be issued with encrypted password enabled.
